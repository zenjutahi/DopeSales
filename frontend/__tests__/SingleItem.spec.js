import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import wait from "waait";
import SingleItem, { SINGLE_ITEM_QUERY } from "../components/SingleItem";
import { MockedProvider } from "react-apollo/test-utils";
import { fakeItem } from "../lib/testUtils";

describe("<SingleItem/>", () => {
  it("render with proper data", async () => {
    const mocks = [
      {
        // when someone makes a request with this query and varibale combo
        request: { query: SINGLE_ITEM_QUERY, variables: { id: "123" } },
        // return this fake data (mocked data)
        result: {
          data: {
            item: fakeItem()
          }
        }
      }
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="123" />
      </MockedProvider>
    );
    expect(wrapper.text()).toContain("Loading....");
    await wait();
    wrapper.update();
    expect(toJSON(wrapper.find("h2"))).toMatchSnapshot();
    expect(toJSON(wrapper.find("img"))).toMatchSnapshot();
    expect(toJSON(wrapper.find("p"))).toMatchSnapshot();
  });

  it("Errors with a No item found", async () => {
    const mocks = [
      {
        // when someone makes a request with this query and varibale combo
        request: { query: SINGLE_ITEM_QUERY, variables: { id: "123" } },
        // return this fake data (mocked data)
        result: {
          errors: [{ message: "No item found for" }]
        }
      }
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="123" />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    const item = wrapper.find('[data-test="graphql-error"]');
    expect(item.text()).toContain("No item found for");
    expect(toJSON(item)).toMatchSnapshot();
  });
});
