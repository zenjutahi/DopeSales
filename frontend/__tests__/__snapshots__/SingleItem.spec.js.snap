// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleItem/> Errors with a No item found 1`] = `
<p
  data-test="graphql-error"
>
  <strong>
    Shoot!
  </strong>
  No item found for
</p>
`;

exports[`<SingleItem/> render with proper data 1`] = `
<h2>
  Viewing 
  dogs are best
</h2>
`;

exports[`<SingleItem/> render with proper data 2`] = `
<img
  alt="dogs are best"
  src="dog.jpg"
/>
`;

exports[`<SingleItem/> render with proper data 3`] = `
<p>
  dogs
</p>
`;
